src/components/EditProduct.tsx

```diff
@@ -146,128 +144,105 @@ class EditProduct extends React.Component<IProps, IState> {
                 );
             });
         } else {
-            this.setState({ initial: false });
+            setInitial(false);
         }
     };
-    renderButtons = (initial: boolean, product: ProductData) => {
-        const invalid = !initial && (this.isInvalid() || this.state.processing);
+
+    const renderButtons = (initial: boolean, product: ProductData) => {
```

Comment:
functions like this with html should be moved to their own component

maybe not for in this MR

Reply:
Agreed